marimo/_runtime/runtime.py

```diff
@@ -609,12 +610,17 @@ def _run_cells_internal(self, cell_ids: set[CellId_t]) -> set[CellId_t]:
                 if formatted_output.traceback is not None:
                     with self._install_execution_context(cell_id):
                         sys.stderr.write(formatted_output.traceback)
+                data_stores = get_context().ui_element_registry._data_stores
+                flat_data_store = {
+                    k: v for data_store in data_stores.values() for k, v in data_store.items()
+                }
```

Comment:
If I understand correctly, this sends all plugin data stores, across all cells. If so: Why are all data stores sent, instead of just ones that are created by this cell and/or referenced in this output?

- Could this be a performance regression? Eg sending tons of data created by multiple cells, even though none of those data are output in this cell.